pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -59,38 +59,31 @@ func applyLocalityWeight(
 	// By setting weights across different localities, it can allow
 	// Envoy to weight assignments across different zones and geographical locations.
 	for _, localityWeightSetting := range distribute {
-		if localityWeightSetting != nil &&
-			util.LocalityMatch(locality, localityWeightSetting.From) {
-			misMatched := map[int]struct{}{}
-			for i := range loadAssignment.Endpoints {
-				misMatched[i] = struct{}{}
-			}
+		if localityWeightSetting != nil && util.LocalityMatch(locality, localityWeightSetting.From) {
+			everMatched := map[int]struct{}{}
 			for locality, weight := range localityWeightSetting.To {
-				// index -> original weight
-				destLocMap := map[int]uint32{}
-				totalWeight := uint32(0)
+				inLocality := map[int]struct{}{}
 				for i, ep := range loadAssignment.Endpoints {
-					if _, exist := misMatched[i]; exist {
-						if util.LocalityMatch(ep.Locality, locality) {
-							delete(misMatched, i)
-							destLocMap[i] = ep.LoadBalancingWeight.Value
-							totalWeight += destLocMap[i]
-						}
+					if util.LocalityMatch(ep.Locality, locality) {
+						inLocality[i] = struct{}{}
+						everMatched[i] = struct{}{}
 					}
 				}
 				// in case wildcard dest matching multi groups of endpoints
 				// the load balancing weight for a locality is divided by the sum of the weights of all localities
-				for index, originalWeight := range destLocMap {
-					weight := float64(originalWeight*weight) / float64(totalWeight)
-					loadAssignment.Endpoints[index].LoadBalancingWeight = &types.UInt32Value{
-						Value: uint32(math.Ceil(weight)),
+				for i := range inLocality {
+					epWeight := float64(weight) / float64(len(inLocality))
```

Comment:
I donot think in this case, we should distribute traffic evenly.  We should also consider its original weight.  

If `If no weights are
	// specificed when locality weighted load balancing is enabled, the cluster is
	// assumed to have a weight of 1.`

Reply:
How/why should we consider the original weight? In what case would the weight ever be set? afaik the only time it is ever set is here.

Reply:
For example we have `region1/zone1/subzon1` endpoints of weight 20, and `region1/zone1/subzon2`  with weight 10. And a failover setting matches the two localities with weight 20. How should we assign the weight between the two localities.

Reply:
Oh right.. I thought this was weights set before calling this function. That makes sense I'll test it out tomorrow. Thanks!

Reply:
Thanks for your test. Locality LB now is an alpha feature.